docs/contributing.rst

```diff
@@ -82,6 +82,36 @@ Writing patches
 
 The better a patch is written, the higher the chances that it'll get accepted and the sooner it will be merged.
 
+Before you can submit a change to the Scrapy code base, you need to:
+
+#. Learn the basic of Git_ and Github_
+
+#. Fork_ the Scrapy repository
+
+#. Clone_ your fork
+
+#. Adding_ & committing changes 
+
+.. _Clone: https://help.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository
+.. _Fork: https://help.github.com/en/github/getting-started-with-github/fork-a-repo
+.. _Git: https://try.github.io/
+.. _GitHub: https://help.github.com/en/github/getting-started-with-github
+.. _Adding: https://help.github.com/en/github/managing-files-in-a-repository/adding-a-file-to-a-repository-using-the-command-line
+
+After the above steps, you will need to install Scrapy in your machine by running::
+
+    pip install .
+
+If the command succeeds, then it means Scrapy is successfully installed.
+
+After installing Scrapy, you will have to create a branch in which you will do your work::
+
+    git checkout -b my-branch-name
```

Comment:
Before branching, we should probably ask users to configure the upstream Scrapy repository as the `upstream` remote in their clone, and then recommend them to create branches the following way:

```bash
git fetch upstream
git checkout upstream/master -b new-branch-name
```

Otherwise, the second time they create a branch, it will probably be based on a previous, different set of changes to Scrapy that may not have been merged yet.